onboarding/source/amb_light_service.c

```diff
@@ -57,34 +56,30 @@ uint8_t initLightService(void) {
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
-    adcData_t adc_data;
-    adcData_t *adc_data_ptr = &adc_data;
+    adcData_t adcData;
+    light_event_t *currentEvent = NULL;
```

Comment:
`xQueueReceive` will copy the event to the address pointed to by `eventReceived`. So `eventReceived` cannot be NULL. It should point to a valid block of memory that you've allocated. I'd suggest just declaring a `light_event_t` variable like `light_event_t event` and passing `&event` into `xQueueReceive`.